src/ts/index.ts

```diff
@@ -0,0 +1,291 @@
+   * fees) still get executed.
+   */
+  tip: BigNumberish;
+  /**
+   * The order kind.
+   */
+  kind: OrderKind;
+  /**
+   * Specifies whether or not the order is partially fillable.
+   */
+  partiallyFillable: boolean;
+}
+
+/**
+ * Gnosis Protocol v2 order flags.
+ */
+export type OrderFlags = Pick<Order, "kind" | "partiallyFillable">;
+
+/**
+ * Order kind.
+ */
+export const enum OrderKind {
+  /**
+   * A sell order.
+   */
+  SELL,
+  /**
+   * A buy order.
+   */
+  BUY,
+}
+
+/**
+ * Reserved nonce value used to indicate that an order can be replayed in any
+ * batch.
+ */
+export const REPLAYABLE_NONCE = 0;
+
+function timestamp(time: number | Date): number {
+  return typeof time === "number" ? time : ~~(time.getTime() / 1000);
+}
+
+function encodeOrderFlags(flags: OrderFlags): number {
+  const kind = flags.kind === OrderKind.SELL ? 0x00 : 0x01;
+  const partiallyFillable = flags.partiallyFillable ? 0x02 : 0x00;
+  return kind | partiallyFillable;
+}
+
+/**
+ * A class for building a buffer of encoded orders.
+ *
+ * The encoder ensures that token addresses are kept track of and performs
+ * necessary computation in order to map each token addresses to IDs.
+ */
+export class OrderEncoder {
+  private readonly _tokens: string[] = [];
+  private readonly _tokenMap: Record<string, number | undefined> = {};
+  private _encodedOrders = "0x";
+
+  /**
+   * Creates a new order encoder instance.
+   * @param domainSeparator Optionally specified domain separator used for
+   * signing orders to encode. See {@link hashOrder} for more details.
+   */
+  public constructor(public readonly domainSeparator?: string) {}
+
+  /**
+   * Gets the array of token addresses used by the currently encoded orders.
+   *
+   * This is used as encoded orders reference tokens by index instead of
+   * directly by address for multiple reasons:
+   * - Reduce encoding size of orders to save on `calldata` gas.
+   * - Direct access to a token's clearing price on settlement instead of
+   *   requiring a search.
+   */
+  public get tokens(): string[] {
+    // NOTE: Make sure to slice the original array, so it cannot be modified
+    // outside of this class.
+    return this._tokens.slice();
+  }
+
+  /**
+   * Gets the encoded orders as a hex-encoded string.
+   */
+  public get encodedOrders(): string {
+    return this._encodedOrders;
+  }
+
+  /**
+   * Gets the number of orders currently encoded.
+   */
+  public get orderCount(): number {
+    // NOTE: Each order is 204 bytes long, so 408 hex characters.
+    return (this._encodedOrders.length - 2) / 408;
+  }
+
+  /**
+   * Returns the signature for the specified order.
+   * @param owner The owner for the order used to sign.
+   * @param domainSeparator The domain separator to add to the digest used for
+   * signing. See {@link hashOrder} for more details.
+   * @param order The order to compute the digest for.
+   * @return Signature for the order.
```

Comment:
The description is not consistent with the parameters of the function.

Reply:
:see_no_evil: